tests/test_automatic_fields.py

```diff
@@ -130,8 +130,25 @@ def test_run(self):
 
                 # verify by values, that we replicated the expected records
                 for actual_record in actual_records:
+                    if not actual_record in expected_records.get(stream):
+                        print("\n==== DATA DISCREPANCY ====\n")
+                        print("Expected: {}\n".format(actual_record))
+                        e_record = [record for record in expected_records.get(stream)
+                                    if actual_record.get('eid') == record.get('eid')]
+                        print("Actual: {}\n".format(e_record))
+                        for key in schema_keys:
+                            e_val = e_record[0].get(key)
+                            val = actual_record.get(key)
+                            if e_val != val:
+                                print("\nDISCREPANCEY | KEY {}: ACTUAL: {} EXPECTED {}".format(key, val, e_val))
```

Comment:
With Adroll, this caught our missing datetime conversions, will this catch them here?

Reply:
It should catch any discrepancies between the json in our schema file, and what is returned by square's api. If there is a missing field, or an incorrect data type it should be caught. I also put these assertions in the bookmarks test as well. The tap was built using the square client so there isn't as much independence between the tap's client and the test_client compared to Adroll. But having these assertions in the bookmarks test where we set additional expectations based off a POST and a PUT should ensure more independence between EXPECTED and ACTUAL.

Reply:
This way of comparing schema isn't perfect though. We are not able to catch the case where the api returns a string but we want to type it as a date-time. Which I am not sure if that is something we should even be doing?